src/Illuminate/Foundation/Testing/DatabaseTransactions.php

```diff
@@ -4,12 +4,36 @@
 
 trait DatabaseTransactions
 {
+    /**
+     * The connections currently affected by the transaction. Indexed by connection name.
+     *
+     * @var array
+     */
+    protected $transacting = [];
+
+    /**
+     * Retrieve the names of the database connections that should be included in the transaction.
+     *
+     * The default value of empty string will cause only the default database connection to be included in the transaction.
+     *
+     * @return array
+     */
+    protected function connectionsToTransact()
+    {
+        return [''];
+    }
+
     public function beginDatabaseTransaction()
     {
-        $this->app->make('db')->beginTransaction();
+        foreach ($this->connectionsToTransact() as $name) {
+            $this->transacting[$name] = $this->app->make('db')->connection($name);
```

Comment:
Might be an idea to resolve `$this->app->make('db')` once outside the loop for performance reasons.